### PR TITLE
MAKE: Enforce no mixing of declarations and code during compilation

### PR DESCRIPTION
**Pull request by gleon99**
## What
Add the "-Werror=declaration-after-statement" flag to compilation command of every file in the repo, and fix incompatible code accordingly.

## Why ?
To enforce at compile-time the "no mixing of declaration & code" rule from our coding guidelines, rather than requiring a human review for that.

### CODE REVIEW

File: src/uct/ib/mlx5/ib_mlx5.inl [hunk 1]
```diff
@@ -285,10 +285,21 @@ uct_ib_mlx5_set_ctrl_seg(struct mlx5_wqe_ctrl_seg* ctrl, uint16_t pi,
...
+                        2,               /* opcode */
+                        14, 13, 12,      /* QP num */
+                        8,               /* data size */
+                        16,              /* signature (set 0) */
+                        16, 16,          /* reserved (set 0) */
+                        0,               /* signal/fence_mode */
+                        16, 16, 16, 16}; /* immediate (set to 0)*/
+    uint32x4_t data = {(opcode << 16) | (opmod << 8) | (uint32_t)fm_ce_se,
+                       pi, ds, qp_num};
+#endif
```
Comment (yosefe) [inline src/uct/ib/mlx5/ib_mlx5.inl:301]: space line after

File: src/uct/ugni/base/ugni_md.c [hunk 2]
```diff
@@ -180,24 +180,22 @@ uct_ugni_md_open(uct_component_h component,const char *md_name,
...
+    md_ops.close              = uct_ugni_md_close;
+    md_ops.query              = uct_ugni_md_query;
+    md_ops.mem_alloc          = (void*)ucs_empty_function;
+    md_ops.mem_free           = (void*)ucs_empty_function;
+    md_ops.mem_reg            = uct_ugni_mem_reg;
+    md_ops.mem_dereg          = uct_ugni_mem_dereg;
+    md_ops.mkey_pack          = uct_ugni_rkey_pack;
+    md_ops.detect_memory_type = ucs_empty_function_return_unsupported;
+
+    md.super.ops       = &md_ops;
```
Comment (yosefe) [inline src/uct/ugni/base/ugni_md.c:196]: align = on column

File: src/uct/ugni/rdma/ugni_rdma_ep.c [hunk 3]
```diff
@@ -13,10 +13,13 @@
 #include <uct/ugni/base/ugni_device.h>
 
 #define UCT_CHECK_PARAM_IOV(_iov, _iovcnt, _buffer, _length, _memh) \
+    void     *_buffer; \
+    size_t    _length; \
+    uct_mem_h _memh; \
```
Comment (yosefe) [inline src/uct/ugni/rdma/ugni_rdma_ep.c:18]: space line ( `   \`)  after

File: src/ucs/profile/profile_on.h [hunk 4]
```diff
@@ -117,9 +117,11 @@ BEGIN_C_DECLS
     \
     _ret_type _name(__VA_ARGS__) { \
         UCS_PROFILE_SCOPE_BEGIN(); \
-        _ret_type _ret = _name##_inner _arglist; \
-        UCS_PROFILE_SCOPE_END(#_name); \
-        return _ret; \
+        { \
+            _ret_type _ret = _name##_inner _arglist; \
```
Comment (yosefe) [inline src/ucs/profile/profile_on.h:121]: maybe just define `_ret_type _ret` in the begining of the function, no need to open new {} scope?

**Review by yosefe — APPROVED** [review]
pls squash (but no rebase)

**Comment by gleon99** [comment]
Squash done

**Pull request merged**